### DISCUSSION TITLE
Record History Creating 2 Entries

### DISCUSSION THREAD

**Discussion by ivojolo** [cortexrd/Knack-Toolkit-Library#421] [Q&A]
Hello, I've tried setting up the _arh and _vrh keywords using the wiki. I also tried referring to Carl's video.

Everything is working, except, I get two entries per instance where I've enabled the _arh keyword. The first record that's created always contains "undefined" on random fields that I can't figure out why it's even referring to those fields, ie:
Invoice (field_2827):
	Before:	undefined
	After:	

Changed By (field_2824):
	Before:	undefined
	After:	

The 2nd record that's created looks correct, ie:
Description (field_1927):
	Before:	2022 Renewals for DEMO CUSTOMER - IMPORTER1
	After:	2022 Renewals for DEMO CUSTOMER - IMPORTER5

I can't figure out what's causing this. I've tried the following:
1. Ensure the Record History object, fields, scene, and form are all following the naming as per the wiki
2. Changing the order of the fields in the object (does this have any impact? I noticed that the order in the CSV file is different than the order mentioned in the Wiki instructions)
3. Tried the _arh keyword on multiple tables and forms

Let me know if I can provide any additional info to help figure this out.

Thank you so much for KTL!

**Comment by cortexrd** [comment]
Hi @ivojolol,

This can be tricky to reproduce and debug for me.  Is it possible for you to give me access to a page with just a bit of data that has the issue?  No need to access the Builder.

Send me the login info at my email: nd@ctrnd.com with the link to the app.  This would accelerate the process a lot.

Thanks for reporting,
Norm

**Reply by ivojolo** [reply]
Hey Norm, I've sent you an email with testing instructions to reproduce the issue in a fresh app copy with only KTL code in the app.

Thanks again.
